### PR TITLE
Use brew in rnn_cell.py

### PR DESCRIPTION
**Pull request by lukeyeager**
Was https://github.com/caffe2/caffe2/pull/1151.

### CODE REVIEW

**Comment by lukeyeager** [comment]
Should probably merge https://github.com/caffe2/caffe2/pull/1184 before testing this.

File: caffe2/python/rnn_cell.py [hunk 1]
```diff
@@ -622,7 +617,8 @@ def _apply(
                     layer_next_states,
                 )
                 if i > 0 and i in self.residual_output_layers:
-                    layer_input = model.net.Sum(
```
Comment (akyrola) [inline caffe2/python/rnn_cell.py:625]: This is failing with error:
  "indentation contains mixed spaces and tabs"

Comment (akyrola) [reply caffe2/python/rnn_cell.py:625]: Otherwise diff looks good

Comment (lukeyeager) [reply caffe2/python/rnn_cell.py:625]: Oops! Sorry about that.

File: caffe2/python/rnn_cell.py [hunk 2]
```diff
@@ -481,12 +478,10 @@ def _prepare_output_sequence(self, model, state_outputs):
...
             with core.NameScope(self.name or ''):
-                output, _ = model.net.Dropout(
+                output = brew.dropout(
+                    model,
                     output,
-                    [
-                        str(output) + '_with_dropout_mask{}'.format(self.mask),
-                        str(output) + '_dropout_mask{}'.format(self.mask),
-                    ],
+                    str(output),
```
Comment (akyrola) [inline caffe2/python/rnn_cell.py:484]: Here was unfortunately a bug gone undetected. The output should not be in-place, that causes problems with gradient generation in some cases.

Comment (akyrola) [reply caffe2/python/rnn_cell.py:484]: Fix is submitted, landing soon

**Pull request closed without merging**